test/PlatformTopoTest.cpp

```diff
@@ -267,7 +268,11 @@ TEST_F(PlatformTopoTest, hsw_num_domain)
     /// @todo when implemented, add tests for each platform
     EXPECT_EQ(0, topo.num_domain(GEOPM_DOMAIN_BOARD_NIC));
     EXPECT_EQ(0, topo.num_domain(GEOPM_DOMAIN_PACKAGE_NIC));
+#ifdef GEOPM_ENABLE_NVML
+    EXPECT_GE(topo.num_domain(GEOPM_DOMAIN_BOARD_ACCELERATOR), 0);
```

Comment:
is GE intended here?  I would expect this to always return 0 anyway without any additional mocks

Reply:
A mocked accelerator topo would allow for this to be tested without invoking the actual NVML bits.

Reply:
PlatformTopoTests with a mocked accelerator slated for a later PR since this bundles nicely with fixes for failures when running on a system with NVML Lib support but without the driver.  Switched for EXPECT_NO_THROW for now.

Reply:
On second thought, a first pass of PlatformTopo using a mocked accelerator topo is now included in the latest commit for review as part of this PR.  This does not resolve all failures that may occur when running on a system with NVML Lib support but without the driver, however it does address the issue above and most of the PlatformTopoTest cases in that situation.